build/circle-test.sh

```diff
@@ -77,8 +77,18 @@ prepare_artifacts() {
         -d "${2}"
       }
 
+      echo "Posting an issue"
       # Generate string of failed tests: 'TestRaftRemoveRace TestChaos TestHoneyBooBoo'
-      FAILEDTESTS=$(grep -oh '^--- FAIL: \w*' "${outdir}/excerpt.txt" | sed -e 's/--- FAIL: //' | tr '\n' ' ')
+      FAILEDTESTS=$(grep -oh '^--- FAIL: \w*' "${outdir}/excerpt.txt" | sed -e 's/--- FAIL: //' | tr '\n' ' ' || true)
+      if [ -z "${FAILEDTESTS}" -a -n "${CIRCLE_TEST_REPORTS-}" ]; then
+        # If a test timed out, try to grab that from the xml. This is already
+        # almost trying too hard to be clever.
+        FAILEDTESTS=$(find "${CIRCLE_TEST_REPORTS}" -type f -iname '*.xml' |
```

Comment:
much better, but I want to get this in before I get on the plane and into the weekend. I'm going to file an issue to myself to switch it over.